nautobot/extras/datasources/git.py

```diff
@@ -282,14 +167,11 @@ def ensure_git_repository(
         head = repo_helper.checkout(from_branch, head)
         if repository_record.current_head != head:
             repository_record.current_head = head
-            # Make sure we don't recursively trigger a new resync of the repository!
-            repository_record.save(trigger_resync=False)
+            repository_record.save()
 
     except Exception as exc:
         if job_result:
```

Comment:
If we let this bubble up to the job the exception will end up in the jobresult.traceback field so this could probably go away?